frontend/src/routes/Bookshelf.jsx

```diff
@@ -22,6 +22,7 @@ function Bookshelf({ backend, user }) {
   }, [lastUpdate, user, backend]);
   const docs = documents?.filter(x => !!x)
     .map(({_id, dc_title, links}) => [_id, dc_title, links]);
+  const uniqueDocs = Array.from(new Map(docs.map(doc => [doc[0], doc])).values());
```

Comment:
It it is not neccessary to create a new constant... Moreover this means that displayedDocs will not benefit from this deduplication.

Keeping the same idea (of using `Map()`) you could squash the lines 23-25 into a single one:

```js
const docs = [
  ...new Map(
    documents?.filter(x => !!x)
      .map(({_id, dc_title, links}) => [_id, [_id, dc_title, links]]
  ).values()
];
```